java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -280,7 +281,7 @@ public int size() {
    */
   public <T extends ValueVector> AddOrGetResult<T> addOrGetVector(FieldType fieldType) {
     boolean created = false;
-    if (vector instanceof ZeroVector) {
+    if (vector instanceof ZeroVector || vector instanceof NullVector) {
```

Comment:
yeah just instance of null vector should do, please add a test too so that we do not break this in future.